test/test_cmake.rb

```diff
@@ -107,6 +107,37 @@ def test_configure_defaults_with_macos
     end
   end
 
+  def test_configure_defaults_with_freebsd
+    recipe = init_recipe
+    recipe.host = 'some-host'
+
+    with_env({ "CC" => nil, "CXX" => nil }) do
+      MiniPortile.stub(:linux?, false) do
+        MiniPortile.stub(:windows?, false) do
+          MiniPortile.stub(:darwin?, false) do
+            MiniPortile.stub(:freebsd?, true) do
```

Comment:
Is there a way to remove some of these stubs?

Reply:
You could instead stub `MiniPortile.target_os` to return a string that is freebsd-ish.

Reply:
Updated to use `with_stubbed_target` instead.